controllers/machinedeletionremediation_controller.go

```diff
@@ -149,6 +149,36 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 
 	log.Info("node-associated machine found", "node", remediation.Name, "machine", machine.GetName())
 
+	// Detect if Node name is expected to change after Machine deletion given
+	// the Platform type. In case of BareMetal platform, the name is NOT
+	// expected to change, for other providers, instead, the name is expected
+	// to change.
+	// The providerID contains the platform type as prefix (e.g. baremetal:///...)
+	var providerID string
+	if providerID, err = getMachineProviderID(machine); err != nil {
+		log.Error(err, "could not get providerID")
+		return ctrl.Result{}, err
+	}
+	if providerID == "" {
+		log.Info("machine does not have ProviderID")
+		return ctrl.Result{}, nil
+	}
+
+	var status metav1.ConditionStatus
+	var permanentNodeDeletionExpectedMsg string
+	if strings.HasPrefix(providerID, "baremetal") {
+		permanentNodeDeletionExpectedMsg = "Node name NOT expected to change after Machine deletion"
+		status = metav1.ConditionFalse
+	} else {
+		permanentNodeDeletionExpectedMsg = "Node name expected to change after Machine deletion"
+		status = metav1.ConditionTrue
+	}
+
+	if updateRequired := r.setPermanentNodeDeletionExpectedCondition(status, remediation); updateRequired {
```

Comment:
nit: maybe use the same message for the condition and logs? So we don't need to keep them aligned at 2 places....

Reply:
ah good idea :+1: